app/adapters/gist-file.js

```diff
@@ -1,7 +1,9 @@
 import ApplicationAdapter from './application';
 
 export default ApplicationAdapter.extend({
+  seq: 0,
+
 	generateIdForRecord: function(store, type, inputProperties) {
-	  return inputProperties.filePath.replace(/\//gi, '.');
+	  return inputProperties.filePath.replace(/\//gi, '.') + "." + this.incrementProperty('seq');
```

Comment:
Instead of allowing duplicate filePath's, I think that we should delay calling `store.createRecord` until after we have prompted for the file name. 

We should:

* Add a `build` (or `buildProperties`) to `ember-cli` service.
* Move main logic out of `generate` into `build`, leaving `generate` as roughly `return this.store.createRecord('gistFile', this.build(type));`.
* Update `addNew` action in `gist/controller.js` to call `build`, then prompt, then createRecord.